### PR TITLE
Create "what is the app for" section & Restructure readme

### PR DESCRIPTION
**Pull request by Farzannajipour**
Restructure readme file and create "what is the app for" section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,64 @@
...
+    - Configuration of the journeys
+    - Configuration of the entitlements
+    - Communication between journeys
+
+
+## Requirements
+For the best experience, these VSCode extensions for Nx and Jest should be installed. Jest runner allows you to easily run the isolated test as you are developing. Nx Console allows you to easily find and run all the possible Nx Commands.
 
 - `nrwl.angular-console`
 - `firsttris.vscode-jest-runner`.
```
Comment (kirillsud) [inline README.md:42]: Maybe better add links to VS marketplace for these extensions?

```suggestion
- [nrwl.angular-console](https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console)
- [firsttris.vscode-jest-runner](https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner).
```

File: README.md [hunk 2]
```diff
@@ -1,26 +1,64 @@
...
+- This example project is a reference implementation to showcase a number of best practices to use when building a new Angular SPA that leverages Backbase components and libraries. The project is regularly updated, so come back often to check for new improvements.
+- This prototype covers:
+    - Authentication in the app
+    - Enable support of the multiple languages
+    - Integration with a theme
+    - Using existing journey installed from 
+    - local NPM repository
+    - Configuration of the journeys
+    - Configuration of the entitlements
+    - Communication between journeys
```
Comment (kirillsud) [inline README.md:35]: Could we add links to the community for some of this points? Or it is a part of a different task that we planned?

Comment (Farzannajipour) [reply README.md:35]: Yes, this one is for another ticket and we will definitely work on it soon.

File: README.md [hunk 3]
```diff
@@ -1,26 +1,64 @@
...
 
-### Prerequisites
+#### Prerequisites
 
 - Go through the [documentation page](https://community.backbase.com/documentation/foundation_angular/latest/authenticate_users) first to avoid any confusion with implementing authentication in the modelless app
 
-### Important Info
+#### Important Info
 
 The code that is used for authentication is not for production purposes, this is the example to understand the concepts lying under the hood.
```
Comment (yenobi) [inline README.md:58]: This can be removed, it is not actual anymore

File: README.md [hunk 4]
```diff
@@ -1,26 +1,64 @@
...
 Do not copy-paste anything related to the authentication to your banking application.
 
-### How to add authentication to your app
+#### How to add authentication to your app
 
 See the example code in the `app.module.ts`, the related `AuthConfig` in the `environment.ts` files, and the `APP_INITIALIZER` provider logic.
```
Comment (yenobi) [inline README.md:63]:
```suggestion
Check the example code in the `app.module.ts`, the related `AuthConfig` in the `environment.ts` files, and the `APP_INITIALIZER` provider logic.
```

File: README.md [hunk 5]
```diff
@@ -33,7 +71,7 @@ Run `ng g @nrwl/angular:app my-app` to generate an application.
 
 When using Nx, you can create multiple applications and libraries in the same workspace.
```
Comment (yenobi) [inline README.md:72]: Looks like missing header here before the NX section to separate it from the authentication.
What do you think about adding the `Developers guide` section, that will unite `generate a library`, `development server`?

**Review by kirillsud — APPROVED** [review]

**Review by yenobi — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1,28 +1,72 @@
 # Golden Sample Angular App
 
-This example project is a reference implementation to showcase a number of best practices to use when building a new Angular SPA that leverages Backbase components and libraries. The project is regularly updated, so come back often to check for new improvements. For the best experience, these VSCode extenstions for Nx and Jest should be installed. Jest runner allows you to easly run isolated test as you are developing. Nx Console allows you to easily find and run all the possible Nx Commands.
+This golden sample provides examples of the code structure, configuration, and best practices to use for developing your  Backbase journey.
```
Comment (tcorral) [inline README.md:3]: What do you mean by "Backbase journey"? Might it not be confusing?

File: README.md [hunk 7]
```diff
@@ -1,28 +1,72 @@
...
+* [Package as a runnable Docker container](#package-as-a-runnable-docker-container)
+* [Further help](#further-help)
+
+
+
+
+
+## What is the app for
+- This project is a reference implementation to showcase best practices to use when building a new Angular SPA for Backbase components and libraries. The project is updated with the latest versions of the tools and libraries used. It shows guidelines about how to develop a web application for the frontend developers.
+- This prototype includes the following:
```
Comment (tcorral) [inline README.md:27]: "includes" => "covers"

File: README.md [hunk 8]
```diff
@@ -1,28 +1,72 @@
...
+    - Using existing journey installed from local NPM repository
+    - Configuring journeys
+    - Configuring entitlements
+    - Communication between journeys
+ 
+ ### What components are part of the app
+  - Auth module
+  - Locale selector for SPA
+  - Theme
+  - Entitlement
```
Comment (tcorral) [inline README.md:40]: "Entitlement" => "Entitlements"

File: README.md [hunk 9]
```diff
@@ -1,28 +1,72 @@
...
+  - Configuration of journeys
+  - Communication service
+  - Simplistic journey examples like transactions and transfer
+
+## Prerequisites
+For the best experience, the following VSCode extensions for Nx and Jest should be installed. 
+
+- [nrwl.angular-console](https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console): to find and run all the possible Nx Commands.J
```
Comment (tcorral) [inline README.md:48]: Is this ".J", at the end of the sentence, a left over or is intentional?

File: README.md [hunk 10]
```diff
@@ -1,28 +1,72 @@
...
+- [firsttris.vscode-jest-runner](https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner): to easily run the isolated test as you are developing. 
+
+
+- On some AWS environments, due to specific WAF configuration, you may haved to use `http://0.0.0.0:4200/` when accessing the app locally, in order to successfully authenticate.
```
Comment (tcorral) [inline README.md:52]: "haved" => "need"

File: README.md [hunk 11]
```diff
@@ -1,28 +1,72 @@
...
+
 
-On some AWS environments, due to specific WAF configuration, you might need to use `http://0.0.0.0:4200/` when accessing the app locally, in order to successfully authenticate.
 
 ## Authentication details
 
-### Prerequisites
+#### Prerequisites
```
Comment (tcorral) [inline README.md:58]: Why switching from H3 to H4?

File: README.md [hunk 12]
```diff
@@ -1,28 +1,72 @@
...
 
 - Go through the [documentation page](https://community.backbase.com/documentation/foundation_angular/latest/authenticate_users) first to avoid any confusion with implementing authentication in the modelless app
```
Comment (tcorral) [inline README.md:60]: What about customers that have never known about our previous developments using model? Does it need any background?

File: README.md [hunk 13]
```diff
@@ -1,28 +1,72 @@
...
 
-### Important Info
+#### Important Info
```
Comment (tcorral) [inline README.md:62]: Why switching from H3 to H4?

File: README.md [hunk 14]
```diff
@@ -1,28 +1,72 @@
...
 
 The code that is used for authentication is not for production purposes, this is the example to understand the concepts lying under the hood.
```
Comment (tcorral) [inline README.md:64]: I would like to emphasise this and next paragraph.

File: README.md [hunk 15]
```diff
@@ -1,28 +1,72 @@
...
 Do not copy-paste anything related to the authentication to your banking application.
```
Comment (tcorral) [inline README.md:65]: I would like to emphasise this and previous paragraph.

File: README.md [hunk 16]
```diff
@@ -1,28 +1,72 @@
...
 
-### How to add authentication to your app
+#### How to add authentication to your app
 
-See the example code in the `app.module.ts`, the related `AuthConfig` in the `environment.ts` files, and the `APP_INITIALIZER` provider logic.
+Check the example code in the `app.module.ts`, the related `AuthConfig` in the `environment.ts` files, and the `APP_INITIALIZER` provider logic.
```
Comment (tcorral) [inline README.md:69]: Can you not add a direct link to the file, and maybe to the lines you want the developer to pay attention to?

File: README.md [hunk 17]
```diff
@@ -33,7 +77,7 @@ Run `ng g @nrwl/angular:app my-app` to generate an application.
 
 When using Nx, you can create multiple applications and libraries in the same workspace.
```
Comment (tcorral) [inline README.md:78]: Add a link to Nx web page in "Nx" word

File: README.md [hunk 18]
```diff
@@ -43,7 +87,7 @@ Run `ng g @nrwl/angular:lib my-lib` to generate a library.
 
 Libraries are shareable across libraries and applications. They can be imported from `@backbase/mylib`.
 
-After generating, use appropiate tags in both `nx.json` and `.eslintrs.json` to impose constraints on the dependency graph.[Nx Tags](https://nx.dev/structure/monorepo-tags)
+After generating, use appropriate tags in both `nx.json` and `.eslintrs.json` to impose constraints on the dependency graph.[Nx Tags](https://nx.dev/structure/monorepo-tags)
```
Comment (tcorral) [inline README.md:90]: Why repeating the same paragraph in two different places so close?

File: README.md [hunk 19]
```diff
@@ -1,13 +1,57 @@
...
+- [nrwl.angular-console](https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console): to find and run all the possible Nx Commands.
+- [firsttris.vscode-jest-runner](https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner): to easily run the isolated test as you are developing. 
+
+
+- On some AWS environments, due to specific WAF configuration, you may need to use `http://0.0.0.0:4200/` when accessing the app locally, in order to successfully authenticate.
+
 
-On some AWS environments, due to specific WAF configuration, you might need to use `http://0.0.0.0:4200/` when accessing the app locally, in order to successfully authenticate.
 
 ## Authentication details
```
Comment (Ela-ch) [inline README.md:56]:
```suggestion
## Authentication
```
Add a short sentence on authentication wrt to the app and include the name of the files where this is defined: - `app.module.ts` 
- `AuthConfig` in `environment.ts`
- `APP_INITIALIZER`
and include the links. 
Add what cannot be put as a comment to the files here - e.g. sequence, specific requirements, or warning note of what they should not do in this particular case.

File: README.md [hunk 20]
```diff
@@ -22,7 +66,7 @@ Do not copy-paste anything related to the authentication to your banking applica
 
 ### How to add authentication to your app
```
Comment (Ela-ch) [inline README.md:67]: Make this title into text with the details of where they can do this.

File: README.md (hunk 16 above)
Comment (Ela-ch) [reply README.md:69]: Agree, See my comment above

File: README.md [hunk 21]
```diff
@@ -22,7 +66,7 @@ Do not copy-paste anything related to the authentication to your banking applica
...
 
-See the example code in the `app.module.ts`, the related `AuthConfig` in the `environment.ts` files, and the `APP_INITIALIZER` provider logic.
+Check the example code in the `app.module.ts`, the related `AuthConfig` in the `environment.ts` files, and the `APP_INITIALIZER` provider logic.
 Secure routes with `AuthGuard`s. We rely on <https://github.com/manfredsteyer/angular-oauth2-oidc>, check their documentation for more details.
```
Comment (Ela-ch) [inline README.md:70]: Is this a recommendation? 
If yes, the state that in the bulleted list for authentication above and provide the link. Not necessary to have text about checking documentation - you have provided the link.

File: README.md [hunk 22]
```diff
@@ -31,9 +75,9 @@ Run `ng g @nrwl/angular:app my-app` to generate an application.
 
 > You can also use Nx Console to generate libraries as well.
 
-When using Nx, you can create multiple applications and libraries in the same workspace.
+When using When using [Nx](https://nx.dev/), you can create multiple applications and libraries in the same workspace.
```
Comment (Ela-ch) [inline README.md:78]:
```suggestion
When using [Nx](https://nx.dev/), you can create multiple applications and libraries in the same workspace.
```

File: README.md [hunk 23]
```diff
@@ -31,9 +75,9 @@ Run `ng g @nrwl/angular:app my-app` to generate an application.
...
 
-After generating, use appropiate tags in both `nx.json` and `.eslintrs.json` to impose constraints on the dependency graph. [Nx Tags](https://nx.dev/structure/monorepo-tags)
+After generating, use appropriate tags in both `nx.json` and `.eslintrs.json` to impose constraints on the dependency graph. [Nx Tags](https://nx.dev/structure/monorepo-tags)
```
Comment (Ela-ch) [inline README.md:80]:
```suggestion
After the app has been generated, use tags in `nx.json` and `.eslintrs.json` to impose constraints on the dependency graph. [Nx Tags](https://nx.dev/structure/monorepo-tags)
```

File: README.md [hunk 24]
```diff
@@ -43,7 +87,6 @@ Run `ng g @nrwl/angular:lib my-lib` to generate a library.
 
 Libraries are shareable across libraries and applications. They can be imported from `@backbase/mylib`.
```
Comment (Ela-ch) [inline README.md:88]:
```suggestion
Libraries can be shared across libraries and applications. You can import them from `@backbase/mylib`.
```
It is not clear what you want to say with the second sentence. Do you mean: "you can import other libraries from  `@backbase/mylib` for other apps?
Or do you mean that you can access other libraries, and "share" them (i.e. means not having a local version of the library for the app?"

File: README.md [hunk 25]
```diff
@@ -43,7 +87,6 @@ Run `ng g @nrwl/angular:lib my-lib` to generate a library.
...
 
-After generating, use appropiate tags in both `nx.json` and `.eslintrs.json` to impose constraints on the dependency graph.[Nx Tags](https://nx.dev/structure/monorepo-tags)
 
 ## Development server
```
Comment (Ela-ch) [inline README.md:91]:
```suggestion
## Load app on a development server
```
Is this what you mean here?

File: README.md [hunk 26]
```diff
@@ -57,13 +100,15 @@ Run `ng g component my-component --project=my-app` to generate a new component.
 
 Run `ng build my-app` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.
```
Comment (Ela-ch) [inline README.md:101]:
```suggestion
To run the project on a development server, run 
`ng build my-app

The build artifacts are stored in the `dist/` directory. 

To build the app to production, use the `--prod` flag.
```
The title for this section appears to be for the development server. Should this be a separate title for production?
Perhaps you can have this as a single title: 
Building your app

and the different commands in for the different build environments.

File: README.md [hunk 27]
```diff
@@ -57,13 +100,15 @@ Run `ng g component my-component --project=my-app` to generate a new component.
...
 
-## Running unit tests
+## Tests
+
+### Running unit tests
```
Comment (Ela-ch) [inline README.md:105]:
```suggestion
- Running unit tests
```
Propose that you have a single title: Testing
and then you have bulleted items or in a table: the different commands
Keep it simple.

File: README.md [hunk 28]
```diff
@@ -57,13 +100,15 @@ Run `ng g component my-component --project=my-app` to generate a new component.
...
 
 Run `ng test my-app` to execute the unit tests via [Jest](https://jestjs.io).
 
 Run `nx affected:test` to execute the unit tests affected by a change.
 
-## Running end-to-end tests
+### Running end-to-end tests
```
Comment (Ela-ch) [inline README.md:111]:
```suggestion
-  Running end-to-end tests
```

File: README.md [hunk 29]
```diff
@@ -1,13 +1,57 @@
...
+* [Generate an application](#generate-an-application)
+* [Generate a library](#generate-a-library)
+* [Development server](#development-server)
+* [Code scaffolding](#code-scaffolding)
+* [Build](#build)
+* [Tests](#tests)
+* [Understand your workspace](#understand-your-workspace)
+* [Running with docker](#running-with-docker)
+* [Package as a runnable Docker container](#package-as-a-runnable-docker-container)
+* [Further help](#further-help)
```
Comment (Ela-ch) [inline README.md:19]: I do not see all of these sections - do they still have to be completed?

**Review by Ela-ch — COMMENTED** [review]
I have done a first pass of the READ ME.
Reduce the sections to overall commands and info that they need to understand and use the app. 
I propose a second review to identify what is necessary and what can be included in the code, and what they can find in the documentation.

File: README.md (hunk 29 above)
Comment (Farzannajipour) [reply README.md:19]: All the sections are already covered.

**Pull request merged**